sdk/resourcemanager/docs/THREAD_POOL_AND_CONNECTION_POOL.md

```diff
@@ -0,0 +1,90 @@
+            .evictInBackground(Duration.ofSeconds(120)) // Every two minutes, the connection pool is regularly checked for connections that are applicable for removal.
+            .build());
+```
+
+### <a name="httpthreadpool"></a>Configure thread pool for the singleton HttpClient:
+By default the TCP client uses an “Event Loop Group”, where the number of the worker threads equals the number of processors available to the runtime on initialization (but with a minimum value of 4). When you need a different configuration, you can use one of the LoopResource#create methods.
+```java readme-sample-azureClientThreadPool
+// Thread pool configuration.
+// Official Reactor Netty documentation for defaults: https://projectreactor.io/docs/netty/release/reference/#client-tcp-level-configurations-event-loop-group
+singletonHttpClientBuilder
+    .eventLoopGroup(LoopResources
+        .create(
+            "client-thread-pool", // thread pool name
+            Runtime.getRuntime().availableProcessors() * 2, // thread pool size
+            true)
+        // we use our custom event loop here, disable the native one
+        .onClient(false))
+    .build();
+```
+
+### <a name="azureidentity"></a>Configure Azure Identity client to use the singleton HttpClient (DefaultAzureCredential for example):
+By default, Azure Identity uses `ForkJoinPool.commonPool()` for token acquisition. The pool size equals the number of processors available to the runtime on initialization minus 1 (with a minimum of 1). It's a singleton pool and is shared by default. Default pool configuration should be sufficient for most use cases.
+```java readme-sample-azureIdentityThreadpool
+// Use the singleton httpClient for Azure Identity
+final AzureProfile profile = new AzureProfile(AzureEnvironment.AZURE);
+final TokenCredential credential = new DefaultAzureCredentialBuilder()
+    .authorityHost(profile.getEnvironment().getActiveDirectoryEndpoint())
+    .executorService(ForkJoinPool.commonPool()) // thread pool for executing token acquisition, usually we leave it default
+    .httpClient(singletonHttpClient)
+    .build();
+```
+
+### <a name="azureclient"></a>Configure Azure Clients to use the singleton HttpClient and Azure Identity client:
+```java readme-sample-azureClientHttpClient
+// Use the singleton httpClient for your Azure client
+AzureResourceManager azureResourceManager = AzureResourceManager
+    .configure()
+    .withLogLevel(HttpLogDetailLevel.BASIC)
+    .withHttpClient(singletonHttpClient)
+    .authenticate(credential, profile)
+    .withSubscription(subscriptionId); // your subscription ID, can be different for different Azure clients
+```
+
+### <a name="reactor"></a>Configure Reactor thread pool size:
+By default, Azure Client subscribes on Schedulers.parallel(), which has a thread pool of size equal to available processor count. It is a static pool and is shared in nature.
```

Comment:
Does the management client use a specific scheduler by default? At least in Core there isn't any explicit configurations of which Scheduler is being used

Reply:
In [Replacing Default Schedulers](https://projectreactor.io/docs/core/release/reference/#scheduler-factory), Schedulers are introduced by operators that need Schedulers for parallelism. 

I'll check where we introduced these operators.

Reply:
I tried replacing Netty implementation to Azure Core OkHttp, and no `parallel-x` threads showed up ever.

Also in [this thread](https://stackoverflow.com/questions/71925290/what-is-the-default-scheduler-in-project-reactor), one claims that `Reactor Netty` would initiate subscription on `Schedulers.parallel()`, though I can't find official docs to back it up.

Guess for now it's safe to assume `Reactor Netty` introduced `Schedulers.parallel()`. Updated docs on this too.

Reply:
Does LRO introduce `Schedulers.parallel()`? I thought Reactor delay (time related operator) would uses it?

You can call the client API directly, they should not have explicit subcribeOn.

I kind of expect client LRO API would create `Schedulers.parallel()` on any HttpClient, but simple client GET API might or might not (depends on which thread pool HttpClient runs on?).

Only a guess. could be wrong.

Reply:
Right, `Flux.delayElement` and `Flux.interval` will specifically introduce `Schedulers.parallel()`, and some of our code used these operators already..

I'll check with LRO, thanks!

Reply:
Tried LRO and it does introduce `Schedulers.parallel()`. Found that `Mono.delaySubscription` is used in `PollerFlux`:https://github.com/Azure/azure-sdk-for-java/blob/1fd4c2f1c2b121a415656db50e0713b1e9843e32/sdk/core/azure-core/src/main/java/com/azure/core/util/polling/PollerFlux.java#L491

I guess lite will still have `Schedulers.parallel()` as well? Since PollerFlux is also used: https://github.com/Azure/azure-sdk-for-java/blob/1fd4c2f1c2b121a415656db50e0713b1e9843e32/sdk/batch/azure-resourcemanager-batch/src/main/java/com/azure/resourcemanager/batch/implementation/BatchManagementClientImpl.java#L296-L311

Reply:
Yeah, Lite is same for LRO.

I just mean `create` flow in Lite won't do explicitly subscribeOn parrallel. But if it calls LRO, same happens.